src/GapicClientTrait.php

```diff
@@ -362,37 +364,29 @@ private function setClientOptions(array $options)
             'libName',
             'libVersion',
         ]);
-
-        $clientConfig = $options['clientConfig'];
-        if (is_string($clientConfig)) {
-            $clientConfig = json_decode(file_get_contents($clientConfig), true);
-        }
+        $options = new ClientOptions($options);
```

Comment:
Probably we could do:
```php
$options = ($options instanceof ClientOptions ? $options : new ClientOptions($options);
```
This would allow the users to pass the object as is when initiating a new Client.

Reply:
The reason I don't want to do this is that it would require us to lose the `array` typehint for all existing clients.

We could do this for V2 clients only, or we can wait until we drop support for PHP 7.4 and add union types (e.g. `array|ClientOptions`)